app/channel.go

```diff
@@ -632,6 +632,9 @@ func (a *App) RestoreChannel(channel *model.Channel, userId string) (*model.Chan
 			Message:   T("api.channel.restore_channel.unarchived", map[string]interface{}{"Username": user.Username}),
 			Type:      model.POST_CHANNEL_RESTORED,
 			UserId:    userId,
+			Props: model.StringInterface{
+				"username": user.Username,
```

Comment:
I'm going to deffer the decision about this towards @jespino, but this looks bad to me. The username can be updated, and hence the data in `Props` might be outdated.

Reply:
We can investigate it a bit more, probably passing here the userId (here and in all the other places), but for now is consistent with the rest of the implementation. (See for example you have the same problem in `POST_CHANNEL_DELETED` post). My proposal is to create another ticket to handle the specific problem of username changes on this kind of messages, but I don't see including that change as part of this issue.

Reply:
Added the ticket here: https://mattermost.atlassian.net/browse/MM-21711